code/game/machinery/firealarm.dm

```diff
@@ -177,18 +183,36 @@
 	else if(!(my_area?.fire || LAZYLEN(my_area?.active_firelocks)))
 		if(my_area?.fire_detect) //If this is false, someone disabled it. Leave the light missing, a good hint to anyone paying attention.
 			if(is_station_level(z))
-				var/current_level = SSsecurity_level.get_current_level_as_number()
-				. += mutable_appearance(icon, "fire_[current_level]")
 				. += emissive_appearance(icon, "fire_level_e", src, alpha = src.alpha)
-				switch(current_level)
+				switch(SSsecurity_level.get_current_level_as_number())
 					if(SEC_LEVEL_GREEN)
 						set_light(l_color = LIGHT_COLOR_BLUEGREEN)
+						. += mutable_appearance(icon, "fire_green")
 					if(SEC_LEVEL_BLUE)
 						set_light(l_color = LIGHT_COLOR_ELECTRIC_CYAN)
+						. += mutable_appearance(icon, "fire_blue")
 					if(SEC_LEVEL_RED)
 						set_light(l_color = LIGHT_COLOR_FLARE)
+						. += mutable_appearance(icon, "fire_red")
 					if(SEC_LEVEL_DELTA)
 						set_light(l_color = LIGHT_COLOR_INTENSE_RED)
+						. += mutable_appearance(icon, "fire_delta")
+					if(SEC_LEVEL_AMBER)
+						set_light(l_color = LIGHT_COLOR_LAVA)
+						. += mutable_appearance(icon, "fire_amber")
+					if(SEC_LEVEL_YELLOW)
+						set_light(l_color =  LIGHT_COLOR_DIM_YELLOW)
+						. += mutable_appearance(icon, "fire_yellow")
+					if(SEC_LEVEL_LAMBDA)
+						set_light(l_color = LIGHT_COLOR_INTENSE_RED)
+						. += mutable_appearance(icon, "fire_lambda")
+					if(SEC_LEVEL_GAMMA)
+						set_light(l_color = LIGHT_COLOR_FLARE)
+						. += mutable_appearance(icon, "fire_red")
+					if(SEC_LEVEL_EPSILON)
+						set_light(l_color = LIGHT_COLOR_FAINT_BLUE)
+						. += mutable_appearance(icon, "fire_offstation")
```

Comment:
this should prolly be a part of `/datum/security_level`, honestly

won't be picky, still okay with merging this PR without that, but would be preferred if possible